example-library/src/main/java/uk/gov/gchq/palisade/example/config/ExamplePolicyCacheWarmerFactory.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.example.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import uk.gov.gchq.palisade.example.common.ExampleUsers;
+import uk.gov.gchq.palisade.example.hrdatagenerator.types.Employee;
+import uk.gov.gchq.palisade.example.rule.BankDetailsRule;
+import uk.gov.gchq.palisade.example.rule.DutyOfCareRule;
+import uk.gov.gchq.palisade.example.rule.FirstResourceRule;
+import uk.gov.gchq.palisade.example.rule.NationalityRule;
+import uk.gov.gchq.palisade.example.rule.RecordMaskingRule;
+import uk.gov.gchq.palisade.example.rule.ZipCodeMaskingRule;
+import uk.gov.gchq.palisade.service.PolicyCacheWarmerFactory;
+import uk.gov.gchq.palisade.service.request.Policy;
+
+import java.util.Map;
+import java.util.Objects;
+
+@ConfigurationProperties
+public class ExamplePolicyCacheWarmerFactory implements PolicyCacheWarmerFactory {
+
+    private String type;
+    private String owner;
+    private Map<String, String> resourceRules;
+    private Map<String, String> recordRules;
+
+    public ExamplePolicyCacheWarmerFactory() {
+    }
+
+    public ExamplePolicyCacheWarmerFactory(final String type, final String owner, final Map<String, String> resourceRules, final Map<String, String> recordRules) {
+        this.type = type;
+        this.owner = owner;
+        this.resourceRules = resourceRules;
+        this.recordRules = recordRules;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(final String type) {
```

Comment:
setter methods need requireNonNull

Reply:
Used the new code generation templates to resolve this.